src/worker-runner.js

```diff
@@ -32,6 +32,27 @@ export default async function ({
 }) {
   port.postMessage("start");
 
+  const {setupFiles, snapshotSerializers} = test.context.config;
+
+  // https://github.com/facebook/jest/issues/11038
+  for (const setupFile of setupFiles) {
+    const setup = await import(pathToFileURL(setupFile));
+
+    if (setup && setup.default) {
+      await setup.default;
+    }
```

Comment:
@SimenB, should `setupFiles` work like this?

We exposed `setup` function before, https://github.com/prettier/prettier/blob/3cbee742a792413211eea1e4fe2d0dad26ff37ea/tests/config/jest-light-runner/src/worker-runner.js#L30-L36

Reply:
Looking at the Jest source code (https://github.com/facebook/jest/blob/main/packages/jest-repl/src/cli/runtime-cli.ts#L112-L123=), we should do something like this:
```js
const { default: setup } = await import(pathToFileURL(setupFile));
if (typeof setup === "function") await setup();
```
(technically Jest only calls `setup()` if it comes from a CJS file, but it's easier to always call it than to check the file type).

In Prettier's setup file, you could either:
- rewrite it to CJS, use dynamic imports in the `setup` function instead of top-level import declarations, and do `module.exports = setup` _(this also works with Jest's default runner)_;
- export a function named `then` rather than `setup`, because `await import(...)` will automatically call `then` and await it since `{ then: function }` is a promise-like object _(this also works with Jest's default runner)_;
- Use `export default setup` (and not `export default setup()`), so that it will be awaited because it's a function _(this will only work with this runner, since Jest only calls the function for CJS files)_.

Reply:
yeah. we would like to call it for ESM as well, but there's no (supported) way of getting the exported result with `vm` APIIs. So our plan there is to tell people to use TLA for ESM and exporting a function from CJS.